src/lib/services/category-results-service.ts

```diff
@@ -0,0 +1,39 @@
+import __logger from '../logger';
+import HtmlHelper from '../parsers/htmlHelper';
+import { Receiver } from '../../models/receiver';
+
+export interface CategoryResultsService {
+	returnCategoryResultsTable(results: Receiver[], categories: string[]): string
+}
+
+export class DefaultCategoryResultsService implements CategoryResultsService {
+
+	constructor() {}
+
+	returnCategoryResultsTable(results: Receiver[], categories: string[]): string {
+		var tables = '';
+		categories.forEach(category => {
+			tables += `
+					<h2>Category: ${HtmlHelper.uppercaseFirstChar(category)}</h2>
+`;
+			const categoryResults: Receiver[] = this.sortCategoryPegs(results, category);
+			if (categoryResults.length > 0) {
+				tables += HtmlHelper.generateTable(categoryResults);
+			} else {
+				tables +=
+`					<p class="pb-3">There were no pegs given for this keyword</p>`;
+			}
+		});
+		return tables;
+	}
+
+	sortCategoryPegs(results: Receiver[], category: string): Receiver[] {
+		var categoryResults: Receiver[] = results.map(x => Object.assign({}, x)); //deep copy array
```

Comment:
Why deep copy it? On line 35, do

    let categoryResults : Receiver[] = results.filter(result => result.pegs.length > 0);

On line 32, no need to use the copied array, just use the original `results`. Seems weird to call `.forEach` on the copy, only to filter over and assign to the original.

Reply:
If I do that, then the unit tests fail, because the original results array gets modified.

It doesn't work unless I actually copy the array. Arrays are passed by reference, not value, in javascript.